ratis-client/src/main/java/org/apache/ratis/client/api/SnapshotManagementApi.java

```diff
@@ -28,5 +28,15 @@
 public interface SnapshotManagementApi {
 
   /** trigger create snapshot file. */
-  RaftClientReply create(long timeoutMs) throws IOException;
+  default RaftClientReply create(long timeoutMs) throws IOException {
+    return create(0, timeoutMs);
+  }
+
+  /** trigger create snapshot file. If forced, ignore the creation gap. */
+  default RaftClientReply create(boolean force, long timeoutMs) throws IOException {
+    return create(1, timeoutMs);
+  }
+
+  /** trigger create snapshot file if the number of newly applied logs since last snapshot exceeds creationGap. */
```

Comment:
Let's add javadoc for `creationGap` and `@return`:
```java
  /**
   * Trigger to create a snapshot.
   *
   * @param creationGap When (creationGap > 0) and (astAppliedIndex - lastSnapshotIndex < creationGap),
   *                    return lastSnapshotIndex; otherwise, take a new snapshot and then return its index.
   *                    When creationGap == 0, use the server configured value as the creationGap.
   * @return a reply.  When {@link RaftClientReply#isSuccess()} is true,
   *         {@link RaftClientReply#getLogIndex()} is the snapshot index fulfilling the operation.
   */
```